samples/hello-connector-solace/src/main/java/com/solace/quarkus/samples/HelloConsumer.java

```diff
@@ -11,61 +11,56 @@
-                .createPubSubOutboundMetadata();
-        Message<?> outboundMessage = Message.of(p.getPayload(), Metadata.of(outboundMetadata), () -> {
-            CompletableFuture completableFuture = new CompletableFuture();
-            p.ack();
-            completableFuture.complete(null);
-            return completableFuture;
-        }, (throwable) -> {
-            CompletableFuture completableFuture = new CompletableFuture();
-            p.nack(throwable, p.getMetadata());
-            completableFuture.complete(null);
-            return completableFuture;
-        });
-        return outboundMessage;
+    CompletionStage<Void> consumeMessage(SolaceInboundMessage<?> p) {
+        Log.infof("Received message: %s from topic: %s", new String(p.getMessage().getPayloadAsBytes(), StandardCharsets.UTF_8),
+                p.getMessage().getDestinationName());
+        return p.ack();
     }
 
     /**
-     * Publish a simple string from using TryMe in Solace broker and you should see the message published to dynamic destination
-     * topic
+     * Receives message from queue - queue.dynamic.topic and overwrites the topic configured in outgoing channel
+     * dynamic-destination-out
+     *
+     * See [resources/application.properties#mp.messaging.outgoing.dynamic-destination-out.producer.topic]
      *
      * @param p
      */
     @Incoming("dynamic-destination-in")
     @Outgoing("dynamic-destination-out")
     @Acknowledgment(Acknowledgment.Strategy.MANUAL)
     Message<?> consumeAndPublishToDynamicTopic(SolaceInboundMessage<?> p) {
-        Log.infof("Received message: %s", new String(p.getMessage().getPayloadAsBytes(), StandardCharsets.UTF_8));
+        Log.infof("Received message: %s from topic: %s", new String(p.getMessage().getPayloadAsBytes(), StandardCharsets.UTF_8),
+                p.getMessage().getDestinationName());
         SolaceOutboundMetadata outboundMetadata = SolaceOutboundMetadata.builder()
                 .setApplicationMessageId("test")
-                .setDynamicDestination("solace/quarkus/producer/" + p.getMessage().getCorrelationId()) // make sure correlationID is available on incoming message
+                .setDynamicDestination("hello/foobar/" + p.getMessage().getApplicationMessageId())
                 .createPubSubOutboundMetadata();
-        Message<?> outboundMessage = Message.of(p.getPayload(), Metadata.of(outboundMetadata), () -> {
-            CompletableFuture completableFuture = new CompletableFuture();
-            p.ack();
-            completableFuture.complete(null);
-            return completableFuture;
-        }, (throwable) -> {
-            CompletableFuture completableFuture = new CompletableFuture();
-            p.nack(throwable, p.getMetadata());
-            completableFuture.complete(null);
-            return completableFuture;
-        });
+        Message<?> outboundMessage = Message.of(p.getPayload(), Metadata.of(outboundMetadata));
         return outboundMessage;
```

Comment:
```suggestion
        return p.addMetadata(outboundMetadata);
```

Reply:
This is not adding the outboundMetadata to message. I still see inbound metadata while debugging

Reply:
this will relay the incoming message, keeping the payload, existing metadata, ack and nack and adds the outbound metadata you've just build. So when the outbound channel builds the solace message it'll find the outbound metadata, the payload and will trigger the ack of the incoming message when the outbound message is sent. 
You can read of this chain of acknowledgement principal in 
https://smallrye.io/smallrye-reactive-messaging/latest/concepts/acknowledgement/#chain-of-acknowledgment

Reply:
Hi @ozangunalp I have replaced with p.addMetadata(outboundMetadata). However I need to use it as p.addMetadata(outboundMetadata).withAck(() -> p.ack()) to call ack method in incoming channel